src/en/space-station-14/core-tech/proposals/appearancesystem-compliance.md

```diff
@@ -0,0 +1,108 @@
+
+At least, this is how it works in theory. Unfortunately this method of ensuring replicatable visuals has had the following problems:
+- There is no check to prevent game logic systems from editing the sprite directly. That means you can have changes to a sprite that aren't evident from the `AppearanceData`.
+- There is no check for ensuring that game logic is not included in a visualizer system, nor is there one to ensure `OnAppearanceDataChanged` isn't subscribed to outside visualizer systems.
+- There is no way to know what visualizer systems are acting on an entity. Since there is no way to determine what is and isn't a visualizer system, you either have to hard-code which components to copy over, or copy every single component over (including game logic ones).
+
+There are also some minor issue like how there's no consistent naming scheme for visualizer systems (e.g. "SystemNameVisuals" vs. "SystemNameVisualizer"), and how there is a "helper class" named `VisualizerSystem` that is just a normal EntitySystem that requires implementing subscribing to `OnAppearanceDataChanged` but doesn't do anything beyond that.
+
+A good example of this is the Chameleon Projector. It copies the prototype's base sprite, but copying the `AppearanceData` is not sufficient since it also needs to include the component which applies that data. This can be easily seen when copying the appearance of a lathe, as the lathe uses `AppearanceData` and applies the visuals via `MachineSystem`, which means the lathe must have the `MachineComponent` component. Without that component (which also forces the `MachineSystem` functionality onto the entity) it doesn't include the modifications required to accurately copy the sprite. 
+
+![image](https://github.com/user-attachments/assets/77bc4593-c858-4246-b79b-426f530cff10)
+
+*Left*: The base lathe.
+*Right*: The Chameleon Projector copy of the base lathe, with all sprite layers showing. These layers would be hidden via `AppearanceData` and `MachineSystem`, but requires the copy to have `MachineComponent`.
+
+You can hardcode exceptions to this manually, but this is of course not tennable and is bad coding practice when a general solution should be used instead.
+
+### Proposed Solution, Overview
+
+In a perfect world, the process of setting and copying an entity's sprite would be the following:
+
+Setting a sprite: 
+1) An entity is spawned in and has a default prototype with sprite data set; this is effectively the "base" to start from.
+2) A game logic system (`ExampleSystem` & `ExampleComponent`) performs its functions and, as a result, requires the sprite to change.
+3) The game logic system sets a key-value pair in the entity's `AppearanceComponent.AppearanceData`, and ensures the entity has `ExampleVisualizerComponent`. This raises the `OnAppearanceDataChange` event.
```

Comment:
One of my gripes with this is that it will require networking redundant data, and I've generally been pushing people to using data in components instead of relying on AppearanceData unless its a server-side comp, or they want to use something like the `GenericVisualizer`.

I.e., if `ExampleComponent` is a shared/networked component, all the data required to update the sprite is already contained in that component, and copying the data to `AppearanceData` will just mean that anytime the component is modified it has to uneccesarily network that data twice. And currently its even worse as there are no delta states, so it always has to send all of `AppearanceData`.